src/test/coins_tests.cpp

```diff
@@ -211,20 +211,22 @@ BOOST_AUTO_TEST_CASE(coins_cache_simulation_test)
             // Every 100 iterations, flush an intermediate cache
             if (stack.size() > 1 && InsecureRandBool() == 0) {
                 unsigned int flushIndex = InsecureRandRange(stack.size() - 1);
+                if (fake_best_block) stack[flushIndex]->SetBestBlock(InsecureRand256());
```

Comment:
Perhaps we can add a comment about the particular situations in which `fake_best_block` is needed, and why it is needed.

Reply:
Thanks for the look; added a comment.